cmd/mt-backfill/main.go

```diff
@@ -0,0 +1,269 @@
+	if strings.Contains(metric.Name, backfillEnd) {
+		ih.finished <- true
+		return
+	}
+	ih.dHandler.ProcessMetricData(metric, partition)
+	mux.Lock()
+	defer mux.Unlock()
+	lastRcvTime = int64(time.Now().Unix())
+}
+
+// input.Handler interface
+func (ih inputHandler) ProcessMetricPoint(point schema.MetricPoint, format msg.Format, partition int32) {
+	ih.dHandler.ProcessMetricPoint(point, format, partition)
+	mux.Lock()
+	defer mux.Unlock()
+	lastRcvTime = int64(time.Now().Unix())
+}
+
+func main() {
+	log.Infof("metrictank backfilling")
+
+	flag.Parse()
+
+	// logger
+	formatter := &logger.TextFormatter{}
+	formatter.TimestampFormat = "2019-03-21 10:00:00.000"
+	log.SetFormatter(formatter)
+	log.SetLevel(log.InfoLevel)
+
+	// Only try and parse the conf file if it exists
+	path := ""
+	if _, err := os.Stat(*confFile); err == nil {
+		path = *confFile
+	}
+	config, err := globalconf.NewWithOptions(&globalconf.Options{
+		Filename:  path,
+		EnvPrefix: "MT_",
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "configuration file error: %s", err)
+		os.Exit(1)
+	}
+
+	// load configs
+	inKafka.ConfigSetup()
+	memory.ConfigSetup()
+	api.ConfigSetup()
+	cluster.ConfigSetup()
+	mdata.ConfigSetup()
+	cassandraStore.ConfigSetup()
+	config.ParseAll()
+
+	// cluster is required because of aggMetric.add()
+	// this should be configured as single mode
+	api.ConfigProcess()
+	cluster.ConfigProcess()
+	addrParts := strings.Split(api.Addr, ":")
+	port, err := strconv.ParseInt(addrParts[len(addrParts)-1], 10, 64)
+	if err != nil {
+		log.Fatalf("Could not parse port from listenAddr. %s", api.Addr)
+	}
+	cluster.Init("backfill", "none", time.Now(), "http", int(port))
+
+	// other settings
+	inKafka.ConfigProcess("backfill")
+	mdata.ConfigProcess()
+	memory.ConfigProcess()
+
+	// interrupt signals
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// backend store
+	cassandraStore.CliConfig.Enabled = true
+	store, err = cassandraStore.NewCassandraStore(cassandraStore.CliConfig, mdata.TTLs())
+	if err != nil {
+		log.Fatalf("failed to initialize cassandra store. %s", err)
+	}
+
+	// chunk cache
+	ccache := cache.NewCCache()
+
+	// memory store
+	chunkMaxStale := dur.MustParseNDuration("chunk-max-stale", *chunkMaxStaleStr)
+	metricMaxStale := dur.MustParseNDuration("metric-max-stale", *metricMaxStaleStr)
+	gcInterval := time.Duration(dur.MustParseNDuration("gc-interval", *gcIntervalStr)) * time.Second
+	aggMetrics = mdata.NewAggMetrics(store, nil, false, chunkMaxStale, metricMaxStale, gcInterval)
+
+	// input
+	// we use kafkamdm as input
+	inKafka.Enabled = true
+	sarama.Logger = l.New(os.Stdout, "[Sarama] ", l.LstdFlags)
+	inputKafka = inKafka.New()
+
+	// cluster manager
+	cluster.Start()
+
+	if *publicOrg < 0 {
+		log.Fatal("public-org cannot be <0")
+	}
+	idx.OrgIdPublic = uint32(*publicOrg)
+
+	memory.Enabled = true
+	metricIndex = memory.New()
+
+	// api server
+	apiServer, err = api.NewServer()
```

Comment:
Why do we need apiServer?

Reply:
we probably don't need apiServer, will remove this